### PR TITLE
Add testing for pretrained models #28

### PR DESCRIPTION
**Pull request by alexlichtenstein**
# Description

In this pull request, we added a pre-trained model and tests of the resume functionality. In addition, we have provided a training script for 5 classic layouts and updated the documentation.

Fixes # (issue)

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,13 +111,28 @@ You should see all tests passing.
...
+## Training examples
+
+You can also try to train agents for 5 classical layouts
+
+```bash
+(harl_rllib) $ cd ppo
+(harl_rllib) ppo $ bash run_experiments.sh
+```
+
+Please select the correct `num_workers` parameter, otherwise the training of the agents wouldn't be able to start.
```
Comment (micahcarroll) [inline README.md:123]: Maybe rephrase to ~"Choose the `num_workers` parameters to be less than the number of CPUs on your machine." (is that right?)

File: README.md [hunk 2]
```diff
@@ -111,13 +111,28 @@ You should see all tests passing.
...
+
 # Repo Structure Overview
 
 `ppo/`:
 - `ppo_rllib.py`: Primary module where code for training a PPO agent resides. This includes an rllib compatible wrapper on `OvercookedEnv`, utilities for converting rllib `Policy` classes to Overcooked `Agent`s, as well as utility functions and callbacks
 - `ppo_rllib_client.py` Driver code for configuing and launching the training of an agent. More details about usage below
 - `ppo_rllib_from_params_client.py`: train one agent with PPO in Overcooked with variable-MDPs 
 - `ppo_rllib_test.py` Reproducibility tests for local sanity checks
+- `run_experiments.sh` Script for training agents on 5 classical layouts
+- `trained_example/` Pretrained model for testing purposes
```
Comment (micahcarroll) [inline README.md:134]: This is going to be in a Google drive in the end right? Or are you thinking of having one model in the repo itself

File: human_aware_rl/ppo/ppo_rllib_test.py [hunk 3]
```diff
@@ -249,35 +250,63 @@ def test_ppo_fp_sp_yes_phi(self):
...
+            options={'--loglevel': 'ERROR'}
+        ).result
+
+        #Test that the rewards from 1 additional iteration are not too different from the original model
+        #performance
+
+        threshold = 0.1
+
+        with open('trained_example/cramped_room/result.json') as f:
+            j = json.loads(f.readlines()[-1])
```
Comment (micahcarroll) [inline human_aware_rl/ppo/ppo_rllib_test.py:302]: Should probably use a util from `utils.py` to do this (in overcooked-ai or human_aware_rl). If one doesn't exist, we should create a `read_json` util

**Review by micahcarroll — CHANGES_REQUESTED** [review]
Looks pretty good to me! I left a couple of minor things to change!

File: README.md (hunk 1 above)
Comment (alexlichtenstein) [reply README.md:123]: Right, that's how I understood it from training. And simply saying yes. But I've also added additional link to the ray documentation

File: README.md (hunk 2 above)
Comment (alexlichtenstein) [reply README.md:134]: One model stays in the repo itself, to test load, start, stop. Others go to the Google Drive.

File: human_aware_rl/ppo/ppo_rllib_test.py (hunk 3 above)
Comment (alexlichtenstein) [reply human_aware_rl/ppo/ppo_rllib_test.py]: haven't found in documentation any built-in function, so just moved it to utils

**Comment by alexlichtenstein** [comment]
> 5 days ago



> We should probably include the graph of the results we get in the PR too :) Might as well, just so people can cross-check that they're getting similar results

done

**Review by micahcarroll — APPROVED** [review]
LGTM!
